core-api/src/services/access.service.ts

```diff
@@ -0,0 +1,41 @@
+import db from "../config/database";
+import { Access, SafeAccess } from "../entities/access.entity";
+import { ServerError, ErrorCode } from "../utils/serverError";
+import { ValidationError, validate } from "class-validator";
+import Messages from "../utils/messages";
+
+/**
+ * @namespace Services
+ * @class AccessService
+ */
+export class AccessService {
+    /**
+     * Creates a new Access in the database
+     * @param place
+     * @param permission
+     */
+    public static async create(place: Place, permission: Permission): Promise<SafeAccess> {
+        const repository = db.getRepository(Access);
+
+        const newAccess = new Access(place, permission);
+
+        await this.validateFields(newAccess);
+
+        try {
+            await repository.save(newAccess);
+            return newAccess.toSafe();
+        } catch (e) {
+            throw new ServerError(e.message, ErrorCode.DATABASE_CONSTRAINT);
+        }
+    }
+
+    private static async validateFields(newAccess: Access): Promise<void> {
```

Comment:
Remember when you exchange the parameters to ids and create a path to get the Objects that you also must validate if they exists here.

Reply:
Create the messages validation into utils/messages.ts

Reply:
Done, validating only if permission exists since access relation with places is many to one.